qemu/tests/virtio_console.py

```diff
@@ -632,10 +644,20 @@ def _port_replug(device, port_idx):
             # when replugging port from pci to different pci. We should
             # either use symlinks (as in Windows) or replug with the busname
             port = ports[port_idx]
-            vm.monitor.cmd('device_del %s' % port.qemu_id)
+            portdev = get_virtio_port_by_name(vm, port.name)
+            if not portdev:
+                test.error("No port named %s" % port.qemu_id)
+            chardev = portdev.get_param("chardev")
+            port_rid = portdev.get_param("id")
+            vm.devices.simple_unplug(portdev, vm.monitor)
             time.sleep(intr_time)
-            vm.monitor.cmd('device_add %s,id=%s,chardev=dev%s,name=%s'
-                           % (device, port.qemu_id, port.qemu_id, port.name))
+            if not chardev:
+                test.error("No chardev in guest for port %s" % port.qemu_id)
+            ndev = qdevices.QDevice(device)
+            ndev.set_param('id', port_rid)
+            ndev.set_param('chardev', chardev)
+            ndev.set_param('name', port.name)
```

Comment:
Here can be something like:
```
for key, value in {'id': port_rid, 'chardev': chardev, 'name': port.name}.items():
    ndev.set_param(key, value)
```
Both okay to me, up to you.

Reply:
Thanks @vivianQizhu, will update my code.